programs/steward/src/instructions/rebalance.rs

```diff
@@ -163,6 +162,21 @@ pub fn handler(ctx: Context<Rebalance>, validator_list_index: usize) -> Result<(
         validator_stake_info.vote_account_address == validator_history.vote_account,
         StewardError::ValidatorNotInList
     );
+
+    // Checking Vote Account
+    require!(
+        validator_stake_info.vote_account_address == ctx.accounts.vote_account.key(),
+        StewardError::VoteAccountDoesNotMatch
```

Comment:
Could arguably delete the check above if we have this because the validator history address has seeds checking with the vote account

Reply:
Good point